sdk/resourcemanager/Azure.ResourceManager/src/ArmOperation.cs

```diff
@@ -3,13 +3,78 @@
 
 // <auto-generated/>
 
-#nullable disable
+#nullable enable
 
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
 
 namespace Azure.ResourceManager
 {
     /// <inheritdoc/>
-    public abstract class ArmOperation : Operation
-    {
+    public class ArmOperation : Operation
+{
+        private readonly OperationInternal? _operation;
+        private readonly RehydrationToken? _rehydrationToken;
+
+        /// <summary> Initializes a new instance of ArmOperation for mocking. </summary>
+        protected ArmOperation()
+        {
+        }
+
+        internal ArmOperation(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Request request, Response response, OperationFinalStateVia finalStateVia, bool skipApiVersionOverride = false, string? apiVersionOverrideValue = null)
+        {
+            var nextLinkOperation = NextLinkOperationImplementation.Create(pipeline, request.Method, request.Uri.ToUri(), response, finalStateVia, skipApiVersionOverride, apiVersionOverrideValue);
+            _operation = new OperationInternal(nextLinkOperation, clientDiagnostics, response);
+        }
+
+        /// <summary> Create an instance of the <see cref="ArmOperation"/> class from rehydration. </summary>
+        /// <param name="client"> The client parameters to use in these operations. </param>
+        /// <param name="rehydrationToken"> The id of the ArmOperation. </param>
+        public ArmOperation(ArmClient client, RehydrationToken? rehydrationToken)
```

Comment:
I think we want this on `Operation` / `Operation<T>` so that DPG can take advantage of this as well.

@KrzysztofCwalina, @tg-msft what do you guys think?

Reply:
Yes, we want this in data plane too.